pkg/controllers/management/eks/eks_cluster_handler.go

```diff
@@ -281,8 +285,14 @@ func (e *eksOperatorController) onClusterChange(key string, cluster *mgmtv3.Clus
 			cluster, err = e.generateAndSetServiceAccount(cluster)
 			if err != nil {
 				var statusErr error
-				cluster, statusErr = e.setFalse(cluster, apimgmtv3.ClusterConditionWaiting,
-					fmt.Sprintf("failed to communicate with cluster: %v", err))
+				if strings.Contains(err.Error(), fmt.Sprintf(dialer.WaitForAgentError, cluster.Name)) {
+					cluster, statusErr = e.setUnknown(cluster, apimgmtv3.ClusterConditionWaiting, "waiting for cluster agent to be deployed")
+					err = nil
+					e.clusterEnqueueAfter(cluster.Name, enqueueTime)
```

Comment:
Since an error is being returned we shouldn't need to enqueue manually.

Reply:
The error shouldn't be returned in this case. Will refactor to make it more obvious.